### DISCUSSION TITLE
Get a count of items in a list

### DISCUSSION THREAD

**Discussion by stuartcporter** [pnp/pnpcore#928] [Q&A]
hi 
this seems pretty basic - but how are you supposed to get a count of items in a list ?
if i get the list items by paging - I can see Items.Count in the debug window is 6555 - but trying to pull that value from the list.items.count() blows

thanks

**Comment by jansenbe** [comment]
@stuartcporter : it's best to use `IList.ItemCount` property as that way you don't need to first grab all the list items, much faster than the other approaches.

**Reply by stuartcporter** [reply]
unfortunately itemcount includes the folders and files - what if i'm only interested in the number of actual files?
not sure how debug can show the properties but you cannot access it from the object. seems strange to me.

-s

**Reply by jansenbe** [reply]
@stuartcporter : if you only want files you can enumerate all files, but that's still performance sensitive, especially for large libraries as you'll end up with paging and possibly a lot of requests. Iteration model:

```csharp
int count = 0;

// Do a paged retrieval of the list items
await foreach(var item in myList.Items)
{
    if (item.FileSystemObjectType == FileSystemObjectType.File)
    {
        count++;
    }
}
```

I think that using search you can get a refinement based upon content types for a given library, which will allow you to filter out folders without having to iterate over all files. I've used this approach in the Microsoft 365 Assessment tool, see https://github.com/pnp/pnpassessment/blob/main/src/PnP.Scanning/PnP.Scanning.Core/Scanners/Syntex/SyntexScanner.cs#L360-L400 for the code you can reuse.
